app/models/concerns/sentiment_analysis_helper.rb

```diff
@@ -0,0 +1,40 @@
+module SentimentAnalysisHelper
+  extend ActiveSupport::Concern
+
+  def opening_sentiments
+    records = incoming_messages.first(average_message_count)
+    average_sentiment(records)
+  end
+
+  def closing_sentiments
+    records = incoming_messages.last(average_message_count)
+    average_sentiment(records)
+  end
+
+  def average_sentiment(records)
+    {
+      label: average_sentiment_label(records),
+      score: average_sentiment_score(records)
+    }
+  end
+
+  private
+
+  def average_sentiment_label(records)
+    value = records.pluck(:sentiment).sum { |a| a['value'].to_i }
+    value.negative? ? 'negative' : 'positive'
+  end
+
+  def average_sentiment_score(records)
+    total = records.pluck(:sentiment).sum { |a| a['score'].to_f }
+    total / average_message_count
+  end
+
+  def average_message_count
+    incoming_messages.count >= 10 ? 5 : ((incoming_messages.count / 2) - 1)
+  end
```

Comment:
We can just have a fixed number, say 5 messages, it's fine if the messages overlap. Right now, we need to test this with real-world conversations and then tweak the scores

 @sojan-official @pranavrajs WDYT?

Reply:
Sounds good.